drv/sprot-api/README.md

```diff
@@ -0,0 +1,510 @@
+0x00000000 | 00 00 00 00 00 00                               | ......
+$
+```
+
+#### Test sinking various count and size buffers from SP to RoT
+
+##### Success for 10 messages with payload of 10 bytes (4 header + 10 payload)
+```
+slab:~/Oxide/src/hubris,gimletlet$ sp sprot --status
+humility: attached to 0483:374f:0046001F3039510834393838 via ST-Link V3
+subargs=SpRotArgs { send: None, pulse: None, sink: None, status: true, msgtype: "echoreq", timeout: 5000 }
+ops=[Call(TargetFunction(34)), Done]
+empty data: None
+status=Status {
+    supported: 0x00000002,
+    bootrom_crc32: 0x47ae8b8d,
+    rx_invalid: 0x00000001,
+    rx_nop: 0x00000001,
+    rx_overrun: 0x00000000,
+    rx_received: 0x00000003,
+    tx_incomplete: 0x00000000,
+    tx_underrun: 0x00000000,
+}
+slab:~/Oxide/src/hubris,gimletlet$ rot reset
+humility: Opened 1fc9:0143:MMLBHACB24IDJ via CMSIS-DAP
+slab:~/Oxide/src/hubris,gimletlet$ sp reset
+humility: Opened 0483:374f:0046001F3039510834393838 via ST-Link V3
+slab:~/Oxide/src/hubris,gimletlet$ sp sprot -T10000 --sink 200,512
+humility: attached to 0483:374f:0046001F3039510834393838 via ST-Link V3
+subargs=SpRotArgs { send: None, pulse: None, sink: Some("200,512"), status: false, msgtype: "echoreq", timeout: 10000 }
+ops=[Push32(200), Push32(512), Call(TargetFunction(33)), Done]
+empty data: None
+results=[
+    Ok(
+        [
+            0xc8,
+            0x0,
+            0x0,
+            0x0,
+            0x0,
+            0x0,
+            0xf,
+            0x0,
+        ],
+    ),
+]
+slab:~/Oxide/src/hubris,gimletlet$ sp sprot --status
+humility: attached to 0483:374f:0046001F3039510834393838 via ST-Link V3
+subargs=SpRotArgs { send: None, pulse: None, sink: None, status: true, msgtype: "echoreq", timeout: 5000 }
+ops=[Call(TargetFunction(34)), Done]
+empty data: None
+status=Status {
+    supported: 0x00000002,
+    bootrom_crc32: 0x47ae8b8d,
+    rx_invalid: 0x00000000,
+    rx_nop: 0x00000001,
+    rx_overrun: 0x00000012,
+    rx_received: 0x000001af,
+    tx_incomplete: 0x00000000,
+    tx_underrun: 0x00000013,
+}
+```
+
+The ErrorReq/ErrorRsp and SinkReq/SinkRsp messages are not expected to be useful in production and should be conditionally compiled or removed.
+
+The returned SinkRsp message contains a SpRotSinkStatus struct:
+
+```rust
+#[repr(C, packed)]
+pub struct SpRotSinkStatus {
+    pub sent: u16,
+    pub req_crc_err: u16,
+    pub rsp_crc_err: u16,
+    pub flow_err: u16,
+}
+```
+
+So in the above test, 200 messages were sent by the RoT (`SinkRsp`) and there
+were 15 flow errors. There were no CRC errors.
+
+The test code in the SP successfully retried those 15 messages and returned `Ok`
+for the test.
+
+#### Test StatusReq and StatusRsp:
+
+The exact values in the status structure are expected to change and stabilize
+before first customer ship.
+
+/Discussion on contents are appreciated/.
+
+  $ cargo xtask humility $APP -- sprot --status
```

Comment:
Similar to above - I think we can do this via hiffy (`hiffy -c SpRot.status`).

Reply:
Updated. The hiffy calls are now used in the examples. My changes to humility aren't needed. However, humility doesn't deal with read plus write leases, so a change there may be needed.